### PR TITLE
Print multiple validation errors

### PR DESCRIPTION
**Pull request by andronat**
Refers to #7027

Is this on the right track?

I am not really sure how to write tests for this. The `fatal()` function exits with `os.exit(1)`. Which is the best practice to tests those situations?

And secondly, may I change all those if statements to a single `if-else if-else`. statement for better comprehension?

/cc @smarterclayton

### CODE REVIEW

File: pkg/kubectl/cmd/util/helpers.go [hunk 1]
```diff
@@ -56,6 +57,13 @@ func CheckErr(err error) {
 		return
 	}
 
+	if errors.IsInvalid(err) {
+		details := err.(*errors.StatusError).Status().Details
+		prefix := fmt.Sprintf("Validation error in %s: '%s'", details.Kind, details.ID)
+		errs := statusCausesToErrors(details.Causes)
+		fatal(MultipleErrors(prefix, errs))
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/util/helpers.go]: I would use MultilineError here instead

File: pkg/kubectl/cmd/util/helpers.go [hunk 2]
```diff
@@ -56,6 +57,13 @@ func CheckErr(err error) {
 		return
 	}
 
+	if errors.IsInvalid(err) {
+		details := err.(*errors.StatusError).Status().Details
+		prefix := fmt.Sprintf("Validation error in %s: '%s'", details.Kind, details.ID)
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/util/helpers.go]: use %q instead of '%s'

**Comment by smarterclayton** [comment]
Can you add a test case to the utils that creates a single cause error, a multi cause error, and a validation with no causes?

**Comment by andronat** [comment]
Something like this? But still I am not sure how to test `CheckErr()`

File: pkg/kubectl/cmd/util/helpers.go [hunk 3]
```diff
@@ -56,6 +57,13 @@ func CheckErr(err error) {
 		return
 	}
 
+	if errors.IsInvalid(err) {
+		details := err.(*errors.StatusError).Status().Details
+		prefix := fmt.Sprintf("Validation error in %s: %q", details.Kind, details.ID)
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/util/helpers.go]: Can you change this message to "The %s %q is invalid:"

Comment (andronat) [inline pkg/kubectl/cmd/util/helpers.go]: done

**Comment by smarterclayton** [comment]
Your tests are failing

**Comment by andronat** [comment]
Yes, because the `CheckErr()` returns with an `os.exit(1)` and I don't know how to mock(?) or by pass it. Which is the right way in such a case?

**Comment by smarterclayton** [comment]
For now you can just remove that test.  You'd have to change more stuff to make it work, we need to rethink checkErr for testability soon though.

----- Original Message -----

**Comment by andronat** [comment]
Ok, I will remove the tests and I will open a new issue to discuss how can we test checkErr

**Comment by andronat** [comment]
Anything else that needs to be done here?

**Comment by andronat** [comment]
rebased to latest

**Comment by smarterclayton** [comment]
LGTM

**Pull request merged**